docs/CloudFerry_Beginner_Demo_How-To.md

```diff
@@ -0,0 +1,245 @@
+and destination clouds. It's advised that this migration host be a Debian/Ubuntu
+Linux distribution when doing real migrations or any significant testing.
+ 
+The instructions below use the "nfs" VM as the migration host but it could equally
+well be done directly from a Debian/Ubuntu or MacOS workstation. 
+
+It is not recommended to use CentOS as your migration host. 
+
+How-To: CloudFerry for Beginners
+================================
+**System Requirements**
+-----------------------
+
+1.  100GB free disk space as a minimum.
+2.  CPU power to spare 5 cores.
+3.  12GB free RAM (you should have 16GB total as a practical minimum)
+4.  Internet access (for git clone).
+5.  MacOS or Ubuntu Linux (or other Debian distribution) Operating
+    System
+
+**Setting Up CloudFerry Test Environment on MacOS**
+-----------------------
+1.  Download and install Vagrant http://www.vagrantup.com/downloads.html
+
+2.  Download and install VirtualBox https://www.virtualbox.org/wiki/Downloads
+    Open Terminal application and proceed.
+
+3.  [user@MacOS ~/]$ ```mkdir ~/git```
+
+4.  [user@MacOS ~/]$ ```cd ~/git```
+
+5.  [user@MacOS ~/git]$ ```git clone https://github.com/MirantisWorkloadMobility/CloudFerry.git```
+
+6.  [user@MacOS ~/git]$ ```cd ~/git/CloudFerry/devlab```
+
+7.  [user@MacOS ~/git/CloudFerry/devlab]$ ```vagrant up grizzly icehouse juno nfs```
+    
+    Open a new terminal window to watch logs.
+    Copy SSH keys around to make your life easier.
+
+8.  [user@MacOS ~/git/CloudFerry/devlab]$ ```cat ~/.ssh/id_rsa```
+
+9.  [user@MacOS ~/git/CloudFerry/devlab]$ ```cat ~/.ssh/id_rsa.pub```
+
+10. [user@MacOS ~/git/CloudFerry/devlab]$ ```vagrant ssh grizzly```
+
+    a.  [vagrant@grizzly ~/]$ ```vi ~/.ssh/id_rsa```
+        Paste the text from step 8 above.
+
+    b.  [vagrant@grizzly ~/]$ ```chmod 600 ~/.ssh/id_rsa```
+
+    c.  [vagrant@grizzly ~/]$ ```vi ~/.ssh/authorized_keys```
+        Paste in the text from step 9 above.
+
+11. [user@MacOS ~/git/CloudFerry/devlab]$ ```vagrant ssh icehouse```
+
+    a.  [vagrant@icehouse ~/]$ ```vi ~/.ssh/id_rsa```
+        Paste the text from step 8 above.
+
+    b.  [vagrant@icehouse ~/]$ ```chmod 600 ~/.ssh/id_rsa```
+
+    c.  [vagrant@icehouse ~/]$ ```vi ~/.ssh/authorized_keys```
+        Paste in the text from step 9 above.
+
+12. [user@MacOS ~/git/CloudFerry/devlab]$ ```vagrant ssh nfs```
+
+    a.  [vagrant@nfs ~/]$ ```vi ~/.ssh/id_rsa```
+        Paste the text from step 8 above.
+
+    b.  [vagrant@nfs ~/]$ ```chmod 600 ~/.ssh/id_rsa```
+
+    c.  [vagrant@nfs ~/]$ ```vi ~/.ssh/authorized_keys```
+         Paste in the text from step 9 above.
+        
+         Now you've got Vagrant managed VM's running in Virutalbox. 
+        
+         SSH to the nfs VM.
+    d.  [user@MacOS ~/git/CloudFerry/devlab]$ ```vagrant ssh nfs```
+
+13. [vagrant@nfs ~/]$ ```sudo apt-get install virtualbox python-dev python-virtualenv libffi-dev git -y```
```

Comment:
All the commands run on nfs VM can also be run on host machine, which is easier, faster and safe if done within python virtualenv. I would recommend changing this to not using nfs, the primary purpose of nfs VM is to have NFS server there, not to run migration.